my-fund-backend/src/main/java/com/myfund/service/CategoryService.java

```diff
@@ -0,0 +1,115 @@
+package com.myfund.service;
+
+import com.myfund.model.Category;
+import com.myfund.model.DTO.CategoryDTO;
+import com.myfund.model.DTO.CreateCategoryDTO;
+import com.myfund.model.DTO.mapper.CategoryMapper;
+import com.myfund.model.DTO.mapper.SubCategoryMapper;
+import com.myfund.model.SubCategory;
+import com.myfund.model.User;
+import com.myfund.repository.CategoryRepository;
+import com.myfund.repository.SubCategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final SubCategoryRepository subCategoryRepository;
+
+
+    @Autowired
+    public CategoryService(CategoryRepository categoryRepository, SubCategoryRepository subCategoryRepository) {
+        this.categoryRepository = categoryRepository;
+        this.subCategoryRepository = subCategoryRepository;
+    }
+
+
+    public List<CategoryDTO> findAllCategoriesByUser(User user) {
+        return CategoryMapper.categoryListMapToCategoryListDTO(categoryRepository.findAllCategoriesByUser(user));
+
+    }
+
+    public Optional<CategoryDTO> findCategoryByIdAndUser(Long categoryId, User user) {
+        Optional<Category> existingCategoryOpt = categoryRepository.findByIdAndUser(categoryId, user);
+        if (existingCategoryOpt.isPresent()) {
+            return Optional.of(CategoryMapper.categoryMapToCategoryDTO(existingCategoryOpt.get()));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    public CategoryDTO saveCategory(CreateCategoryDTO createCategoryDTO, User user) {
+        Optional<Category> existingCategory = categoryRepository.findByNameAndUser(createCategoryDTO.getName(), user);
+        Category category;
+        if (existingCategory.isPresent()) {
```

Comment:
hmm, it is not clear for me. This method is called SAVECategory. If i pass createCategoryDTO wth already existing category name, I would expect some error/warning instead of returning results of some mapping